### PR TITLE
[RPC] Update GetBlock to align with Core

### PR DESCRIPTION
**Pull request by fassadlr**

### CODE REVIEW

File: src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/BlockSizeRule.cs [hunk 1]
```diff
@@ -39,45 +38,14 @@ public override Task RunAsync(RuleContext context)
             Block block = context.ValidationContext.BlockToValidate;
 
             // Size limits.
-            if ((block.Transactions.Count == 0) || (block.Transactions.Count > options.MaxBlockBaseSize) ||
-                (GetSize(this.Parent.Network, block, TransactionOptions.None) > options.MaxBlockBaseSize))
+            if ((block.Transactions.Count == 0) || (block.Transactions.Count > consensus.Options.MaxBlockBaseSize) ||
```
Comment (zeptin) [inline src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/BlockSizeRule.cs]: I don't really understand how the transaction count could ever be so large. Not that you changed the logic, but it doesn't seem to make sense unless it's possible to have a massive amount of empty transactions somehow.

File: src/Stratis.Bitcoin/Controllers/Models/BlockModel.cs [hunk 2]
```diff
@@ -11,50 +11,87 @@ public class BlockModel
...
 
         [JsonProperty("difficulty")]
         public double Difficulty { get; private set; }
 
-        [JsonProperty("merkleroot")]
-        public string MerkleRoot { get; private set; }
+        [JsonProperty("chainwork")]
+        public string ChainWork { get; private set; }
+
+        [JsonProperty("ntx")]
```
Comment (zeptin) [inline src/Stratis.Bitcoin/Controllers/Models/BlockModel.cs]: This seems to be `nTx` in Core. Would look a bit ugly though.

File: src/Stratis.Bitcoin/Controllers/Models/BlockModel.cs [hunk 3]
```diff
@@ -11,50 +11,87 @@ public class BlockModel
...
+
+        [JsonProperty("height")]
+        public int Height { get; private set; }
+
         [JsonProperty("version")]
         public int Version { get; private set; }
 
-        [JsonProperty("bits")]
-        public string Bits { get; private set; }
+        [JsonProperty("versionhex")]
```
Comment (zeptin) [inline src/Stratis.Bitcoin/Controllers/Models/BlockModel.cs]: `versionHex` in Core

Comment (fassadlr) [reply src/Stratis.Bitcoin/Controllers/Models/BlockModel.cs]: Fixed

File: src/Stratis.Bitcoin/Controllers/Models/BlockModel.cs (hunk 2 above)
Comment (fassadlr) [reply src/Stratis.Bitcoin/Controllers/Models/BlockModel.cs]: Have updated.

File: src/Stratis.Bitcoin/Controllers/Models/BlockModel.cs [hunk 4]
```diff
@@ -11,50 +11,87 @@ public class BlockModel
...
-        [JsonProperty("nonce")]
-        public uint Nonce { get; private set; }
-
-        [JsonProperty("height")]
-        public int Height { get; private set; }
+        [JsonProperty("nextblockhash")]
+        public string NextBlockHash { get; private set; }
 
-        public BlockModel(Block block, ChainBase chain)
+        public BlockModel(Block block, ChainBase chain, Network network, int verbosity = 1)
```
Comment (dangershony) [inline src/Stratis.Bitcoin/Controllers/Models/BlockModel.cs]: Better to pass ChainedHeader in the constructor and also the tip (we will soon delete CC)

**Pull request merged**